src/west/main.py

```diff
@@ -503,16 +509,17 @@ def main(argv=None):
         quote_sh_list(argv))
     try:
         args.handler(args, unknown)
-    except WestUpdated:
-        # West has been automatically updated. Restart ourselves to run the
-        # latest version, with the same arguments that we were given.
+    except WestReload as wr:
+        # West needs to reload itself. Restart ourselves to run the
+        # latest version, with the arguments provided in the exception or
+        # the original ones provided by the user
+        argv = wr.argv or argv
         log.dbg("sys.executable:\"{}\" sys.argv[0]:\"{}\" argv:\"{}\"".format(
                 sys.executable, sys.argv[0], argv))
-        if sys.platform == "win32":
-            # On Windows Python generates .exe executables
-            os.execv(sys.argv[0], argv)
-        else:
-            os.execv(sys.executable, [sys.executable] + [sys.argv[0]] + argv)
+        # Treat the Python script as an executable
+        proc = Popen([sys.argv[0]] + argv)
+        proc.communicate()
+        sys.exit(proc.returncode or 0)
```

Comment:
I had `or errno.EIO` here but apparently for whatever reason `proc.returncode` is sometimes `None` and the unit tests failed. Any ideas?

Reply:
That makes no sense; communicate should wait until the process is done. A None returncode, per documentation, means the subprocess isn't done executing yet.

But please see above about whether this approach is workable.

Reply:
It's strange yes. I will  debug this further and update.

Reply:
Doh, `0 or x` yields x... :|